backend/api/src/main/kotlin/software/shonk/application/service/ShorkService.kt

```diff
@@ -24,6 +24,27 @@ class ShorkService(private val shork: IShork) : ShorkUseCase {
         return Result.success(Unit)
     }
 
+    override fun getLobbySettings(lobbyId: Long): Result<InterpreterSettings> {
+        val lobby =
+            getLobby(lobbyId).getOrElse {
+                return Result.failure(it)
+            }
+        return Result.success(
+            InterpreterSettings(
+                coreSize = lobby.getSettings().coreSize,
+                instructionLimit = lobby.getSettings().instructionLimit,
+                initialInstruction = lobby.getSettings().initialInstruction,
+                maximumTicks = lobby.getSettings().maximumTicks,
+                maximumProcessesPerPlayer = lobby.getSettings().maximumProcessesPerPlayer,
+                readDistance = lobby.getSettings().readDistance,
+                writeDistance = lobby.getSettings().writeDistance,
+                minimumSeparation = lobby.getSettings().minimumSeparation,
+                separation = lobby.getSettings().separation,
+                randomSeparation = lobby.getSettings().randomSeparation,
```

Comment:
It would look much nicer if the settings were bound to a variable and then used via it, instead of repeatedly calling getSettings.

Reply:
you're right, I was intending not to use many variables and make the code less efficient. I found a better implementation that is more efficient than my original one